### PR TITLE
Address staticcheck failures for test/e2e/lifecycle/bootstrap

### PR DESCRIPTION
**Pull request by mattjmcnaughton**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Make small, non-functional changes to make the `test/e2e/lifecycle/bootstrap` pass `staticcheck`.

**Which issue(s) this PR fixes**:
Ref #81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
/assign @spiffxp
/assign @liggitt 

thanks :)

**Review by SataQiu — APPROVED** [review]
/lgtm

File: test/e2e/lifecycle/cluster_upgrade.go [hunk 1]
```diff
@@ -446,9 +446,6 @@ func runUpgradeSuite(
 	upgradeType upgrades.UpgradeType,
 	upgradeFunc func(),
 ) {
-	upgCtx, err := getUpgradeContext(f.ClientSet.Discovery(), *upgradeTarget)
```
Comment (liggitt) [inline test/e2e/lifecycle/cluster_upgrade.go]: getUpgradeContext does sanity checks (server is reachable and reports a parseable version). do we still want to do that just to check the error, e.g.

```go
_, err := getUpgradeContext(f.ClientSet.Discovery(), *upgradeTarget)
framework.ExpectNoError(err)
```

Comment (liggitt) [reply test/e2e/lifecycle/cluster_upgrade.go]: actually, the passed-in upgCtx is not always constructed using this same method, e.g.:

https://github.com/kubernetes/kubernetes/blob/3edbc6afff17ea8dfe5c10b2677dcdc8767f67e2/test/e2e/lifecycle/cluster_upgrade.go#L196

**Comment by liggitt** [comment]
@kubernetes/sig-testing is there a way to run an upgrade test on a PR as a presubmit?

**Comment by BenTheElder** [comment]
upgrade testing has been woefully under-staffed. it might be possible to do kubeadm/kinder upgrade testing.

https://cs.k8s.io/?q=pull-kubernetes-.*upgrade&i=nope&files=&repos=kubernetes/test-infra suggests there isn't one

File: test/e2e/lifecycle/cluster_upgrade.go [hunk 2]
```diff
@@ -446,9 +446,6 @@ func runUpgradeSuite(
 	upgradeType upgrades.UpgradeType,
 	upgradeFunc func(),
 ) {
```
Comment (liggitt) [inline test/e2e/lifecycle/cluster_upgrade.go]: the invocation from here is not identical to what is being removed:
https://github.com/kubernetes/kubernetes/blob/833e8dc10bdce43fc8229c39639a7f7364703874/test/e2e/lifecycle/cluster_upgrade.go#L196-L208

Comment (mattjmcnaughton) [reply test/e2e/lifecycle/cluster_upgrade.go]: Ooo good catch! Going to remove `test/e2e/lifecycle` and just do `test/e2e/lifecycle/bootstrap` in this diff. I'll look into the differences here later.

**Comment by mattjmcnaughton** [comment]
ping @SataQiu @liggitt updated this to only include `test/e2e/lifecycle/bootstrap`. I'll address the other collection of staticcheck failures in a different pr later.

**Pull request merged**